packages/core/src/modules/indy/services/IndyRevocationService.ts

```diff
@@ -0,0 +1,197 @@
+            revocationRegistryId,
+            requestRevocationInterval?.to,
+            0
+          )
+
+          const { tailsLocation, tailsHash } = revocRegDef.value
+          const tails = await this.indyUtilitiesService.downloadTails(tailsHash, tailsLocation)
+
+          const revocationState = await this.indy.createRevocationState(
+            tails,
+            revocRegDef,
+            revocRegDelta,
+            deltaTimestamp,
+            credentialRevocationId
+          )
+          const timestamp = revocationState.timestamp
+
+          if (!revocationStates[revocationRegistryId]) {
+            revocationStates[revocationRegistryId] = {}
+          }
+          revocationStates[revocationRegistryId][timestamp] = revocationState
+        }
+      }
+
+      this.logger.debug(`Created Revocation States for Proof Request`, {
+        revocationStates,
+      })
+
+      return revocationStates
+    } catch (error) {
+      this.logger.error(`Error creating Indy Revocation State for Proof Request`, {
+        error,
+        proofRequest,
+        requestedCredentials,
+      })
+
+      throw isIndyError(error) ? new IndySdkError(error) : error
+    }
+  }
+
+  // Get revocation status for credential (given a from-to)
+  // Note from-to interval details: https://github.com/hyperledger/indy-hipe/blob/master/text/0011-cred-revocation/README.md#indy-node-revocation-registry-intervals
+  public async getRevocationStatus(
+    credentialRevocationId: string,
+    revocationRegistryDefinitionId: string,
+    requestRevocationInterval: RevocationInterval
+  ): Promise<{ revoked: boolean; deltaTimestamp: number }> {
+    this.logger.trace(
+      `Fetching Credential Revocation Status for Credential Revocation Id '${credentialRevocationId}' with revocation interval with to '${requestRevocationInterval.to}' & from '${requestRevocationInterval.from}'`
+    )
+
+    this.assertRevocationInterval(requestRevocationInterval)
+
+    const { revocRegDelta, deltaTimestamp } = await this.ledgerService.getRevocationRegistryDelta(
+      revocationRegistryDefinitionId,
+      requestRevocationInterval.to,
+      0
+    )
+
+    const revoked: boolean = revocRegDelta.value.revoked?.includes(parseInt(credentialRevocationId)) || false
+    this.logger.trace(
+      `Credental with Credential Revocation Id '${credentialRevocationId}' is ${
+        revoked ? '' : 'not '
+      }revoked with revocation interval with to '${requestRevocationInterval.to}' & from '${
+        requestRevocationInterval.from
+      }'`
+    )
+
+    return {
+      revoked,
+      deltaTimestamp,
+    }
+  }
+
+  // TODO: Add Test
+  // Check revocation interval in accordance with https://github.com/hyperledger/aries-rfcs/blob/main/concepts/0441-present-proof-best-practices/README.md#semantics-of-non-revocation-interval-endpoints
+  private assertRevocationInterval(requestRevocationInterval: RevocationInterval) {
```

Comment:
RFC 0441 defines best practices. Should we throw errors if those best practices are not followed?

Reply:
Currently as the code is written it will throw errors if the RFC is not followed. 
But, ff your question was more along the lines of whether we should throw errors vs warnings, I would say errors based off of the inclusion of the 441 RFC in the [AIPv2 INDYCREDS sub-target](https://github.com/hyperledger/aries-rfcs/tree/main/concepts/0302-aries-interop-profile#indycred-indy-based-credentials). I'd also say that there's behavior that may be unexpected/not thought through if the intervals don't follow the RFC, so I'd be concerned with behavior past the assertion. However I'd be more than happy to discuss further, but that's my initial thoughts :)